packages/lesswrong/server/repos/DatabaseMetadataRepo.ts

```diff
@@ -55,4 +56,47 @@ export default class DatabaseMetadataRepo extends AbstractRepo<"DatabaseMetadata
       createdAt: new Date(),
     });
   }
+
+  private electionNameToMetadataName(electionName: string): string {
+    return `${electionName}Hearts`;
+  }
+
+  async getGivingSeasonHearts(electionName: string): Promise<GivingSeasonHeart[]> {
+    const metadataName = this.electionNameToMetadataName(electionName);
+    const result = await this.getRawDb().oneOrNone(`
+      -- DatabaseMetadataRepo.getGivingSeasonHearts
+      SELECT ARRAY_AGG(q."value" || JSONB_BUILD_OBJECT(
+        'userId', u."_id",
+        'displayName', u."displayName"
+      )) "hearts"
+      FROM (
+        SELECT (JSONB_EACH("value")).*
+        FROM "DatabaseMetadata"
+        WHERE "name" = $1
+      ) q
+      JOIN "Users" u ON q."key" = u."_id"
+    `, [metadataName]);
+    return result?.hearts ?? [];
+  }
+
+  async addGivingSeasonHeart(
+    electionName: string,
+    userId: string,
+    x: number,
+    y: number,
+    theta: number,
+  ): Promise<GivingSeasonHeart[]> {
+    const metadataName = this.electionNameToMetadataName(electionName);
+    await this.none(`
+      -- DatabaseMetadataRepo.addGivingSeasonHeart
+      INSERT INTO "DatabaseMetadata" ("_id", "name", "value", "createdAt")
+      VALUES ($1, $2, $3, CURRENT_TIMESTAMP)
+      ON CONFLICT ("name") DO UPDATE SET "value" = "DatabaseMetadata"."value" || $3
+    `, [
+      randomId(),
+      metadataName,
+      {[userId]: {x, y, theta}},
+    ]);
```

Comment:
Interesting - seems fine but please have a plan to clean up the code and the db afterwards, I would prefer we don't leave things in random places